monitor/uidmonitor/processor.go

```
@@ -0,0 +1,398 @@
+
+		if err := s.contextStore.GetContextInfo("/"+contextID, &storedPU); err != nil {
+			continue
+		}
+		eventInfo := storedPU.EventInfo
+		mark := storedPU.MarkVal
+		if pids, ok := marktoPID[mark]; !ok {
+			//No pids with stored mark destroy the context record and go to next context
+			s.contextStore.RemoveContext("/" + contextID)
+			continue
+		} else {
+			for _, pid := range pids {
+				eventInfo.PID = pid
+				s.Start(eventInfo)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+// generateContextID creates the contextID from the event information
+func (s *UIDProcessor) generateContextID(eventInfo *rpcmonitor.EventInfo) (string, error) {
+
+	contextID := eventInfo.PUID
+	if eventInfo.Cgroup != "" {
+		if !s.regStop.Match([]byte(eventInfo.Cgroup)) {
+			return "", fmt.Errorf("Invalid PUID %s", eventInfo.Cgroup)
+		}
+		contextID = eventInfo.Cgroup[strings.LastIndex(eventInfo.Cgroup, "/")+1:]
+	}
+
+	return contextID, nil
+}
+
+func (s *UIDProcessor) processLinuxServiceStart(event *rpcmonitor.EventInfo, runtimeInfo *policy.PURuntime) error {
+	// list, err := cgnetcls.ListCgroupProcesses(event.PUID)
+	// if err == nil {
+	// 	//cgroup exists and pid might be a member
+	// 	isrestart := func() bool {
+	// 		for _, element := range list {
+	// 			if element == event.PID {
+	// 				//pid is already there it is restart
+	// 				return true
+	// 			}
+	// 		}
+	// 		return false
+	// 	}()
+
+	// 	if !isrestart {
+	// 		pid, _ := strconv.Atoi(event.PID)
+	// 		s.netcls.AddProcess(event.PID, pid) // nolint
+	// 		return nil
+	// 	}
+	// }
+
+	//It is okay to launch this so let us create a cgroup for it
+	err := s.netcls.Creategroup(event.PID)
+	if err != nil {
+		return err
+	}
+
+	markval := runtimeInfo.Options().CgroupMark
+	if markval == "" {
+		if derr := s.netcls.DeleteCgroup(event.PID); derr != nil {
+			zap.L().Warn("Failed to clean cgroup", zap.Error(derr))
+		}
+		return errors.New("Mark value not found")
+	}
+
+	mark, _ := strconv.ParseUint(markval, 10, 32)
```

don’t ignore the error